convert2rhel/cert.py

```diff
@@ -20,56 +20,99 @@
-        self._target_cert_dir = "/etc/pki/product-default/"
-        self._cert_filename, self._source_cert_dir = self._get_cert()
+class PEMCert(backup.RestorableChange):
+    """Handling certificates needed for verifying Red Hat services."""
+
+    def __init__(self, source_cert_dir, target_cert_dir):
+        super(PEMCert, self).__init__()
+
+        self._target_cert_dir = target_cert_dir
+        self._source_cert_dir = source_cert_dir
+        self._cert_filename = _get_cert(source_cert_dir)
         self._source_cert_path = self._get_source_cert_path()
         self._target_cert_path = self._get_target_cert_path()
 
-    @staticmethod
-    def _get_cert():
-        """Return name of certificate and this directory."""
-        cert_dir = os.path.join(utils.DATA_DIR, "rhel-certs")
-        if not os.access(cert_dir, os.R_OK | os.X_OK):
-            loggerinst.critical("Error: Could not access %s." % cert_dir)
-        pem_filename = None
-        for filename in os.listdir(cert_dir):
-            if filename.endswith(".pem"):
-                pem_filename = filename
-                break
-        if not pem_filename:
-            loggerinst.critical("Error: System certificate (.pem) not found in %s." % cert_dir)
-        return pem_filename, cert_dir
+        self.previously_installed = False
 
     def _get_source_cert_path(self):
         return os.path.join(self._source_cert_dir, self._cert_filename)
 
     def _get_target_cert_path(self):
         return os.path.join(self._target_cert_dir, self._cert_filename)
 
-    def install(self):
+    def enable(self):
         """RHEL certificate (.pem) is used by subscription-manager to
         determine the running system type/version.
         """
-        try:
-            utils.mkdir_p(self._target_cert_dir)
-            shutil.copy(self._source_cert_path, self._target_cert_dir)
-        except OSError as err:
-            loggerinst.critical("OSError({0}): {1}".format(err.errno, err.strerror))
+        if self.enabled:
+            return
 
-        loggerinst.info("Certificate %s copied to %s." % (self._cert_filename, self._target_cert_dir))
+        if os.path.exists(self._target_cert_path):
+            loggerinst.info("Certificate already present at %s. Skipping copy." % self._target_cert_path)
+            self.previously_installed = True
+        else:
+            try:
+                utils.mkdir_p(self._target_cert_dir)
+                shutil.copy2(self._source_cert_path, self._target_cert_dir)
+            except OSError as err:
+                loggerinst.critical("OSError({0}): {1}".format(err.errno, err.strerror))
 
-    def remove(self):
+            loggerinst.info("Certificate %s copied to %s." % (self._cert_filename, self._target_cert_dir))
+
+        super(PEMCert, self).enable()
+
+    def restore(self):
         """Remove certificate (.pem), which was copied to system's cert dir."""
         loggerinst.task("Rollback: Remove installed RHSM certificate")
```

Comment:
```suggestion
        loggerinst.task("Rollback: Remove installed certificate")
```
This method restores not only the RHSM cert:
```
[2023-09-01T18:59:26+0000] TASK - [Rollback: Remove installed RHSM certificate] **********************
[2023-09-01T18:59:26+0000] DEBUG - Calling command 'rpm -qf /etc/rhsm/ca/redhat-uep.pem'
```